java/ql/src/semmle/code/java/dataflow/ModulusAnalysis.qll

```diff
@@ -72,7 +58,7 @@ private Expr modExpr(Expr arg, int mod) {
   exists(ConstantIntegerExpr c |
     mod = 2.pow([1 .. 30]) and
     c.getIntValue() = mod - 1 and
-    result.(AndBitwiseExpr).hasOperands(arg, c)
+    result.(SimpleBitwiseAndExpr).hasOperands(arg, c)
```

Comment:
I am not sure why we need both `SimpleBitwiseAndExpr` and `BitwiseAndExpr`; perhaps making the fix in `valueFlowStep` removes the need for both.

Reply:
Java was not handling the `x &= 2^n-1` case, and I mechanically converted these to be 1:1 equivalent with previous implementation.